azerothcore/azerothcore-wotlk
# Core/Spells: Added helper to get clean and unmitigated damage for pro…

…c system.

Fixed #6900



## Changes Proposed:
-  Added helper to get clean and unmitigated damage for proc system
-  Fixed damage of Blade Flurry.
-  Fixed damage of Sweeping Strikes.

**Eluna buildifx handled [here](https://github.com/ElunaLuaEngine/Eluna/pull/381)**

## Issues Addressed:

- Closes #6900
- Closes #7636
- Closes https://github.com/chromiecraft/chromiecraft/issues/1167
- Closes https://github.com/chromiecraft/chromiecraft/issues/1580

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

1. get on a atleast level 30 Rogue with the Talent Blade Flurry specced
2. pull 2 mobs with near the same Armor Values and hit them while Blade Flurry is activated
3. notice the second target gets significantly less damage than the first


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
